src/hotspot/os/linux/os_linux.cpp

```diff
@@ -217,6 +220,13 @@ julong os::available_memory() {
   return Linux::available_memory();
 }
 
+int os::Linux::malloc_info(FILE* stream) {
+  if (::malloc_info) {
```

Comment:
This looks strange as an existence check for a library function. At build time what does this get compiled into such that it runtime it can determine whether `malloc_info` exists or not ??

(also style rules say no implicit booleans so check against nullptr).

Reply:
I agree, this looks weird. Please do as the Romans do (in this case, as we do with mallinfo/mallinfo2()). Just follow the breadcrumbs.

Additionally, I would put this alongside os::Linux::get_mallinfo and guard it with ifdef GLIBC. You don't need this on muslc.